src/libraries/System.Memory/tests/ArrayBufferWriter/ArrayBufferWriterTests.T.cs

```diff
@@ -205,6 +205,15 @@ public void GetMemory_DefaultCtor(int sizeHint)
             Assert.Equal(sizeHint <= 256 ? 256 : sizeHint, memory.Length);
         }
 
+        [Fact]
+        public void GetMemory_ExceedMaximumBufferSize()
+        {
+            var output = new ArrayBufferWriter<T>(int.MaxValue / 2 + 1);
+            output.Advance(int.MaxValue / 2 + 1);
+            output.GetMemory(1);//Validate when can't double the buffer size, but can grow by sizeHint
```

Comment:
Also, typo in comment:
```suggestion
            output.GetMemory(1); // Validate we can't double the buffer size, but can grow by sizeHint
```